### PR TITLE
r/elastic_transcoder_pipeline - add validations

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #11872, #13820

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource_aws_elastic_transcoder_pipeline: add plan time validations `content_config.storage_class`, `content_config_permissions.access`, `content_config_permissions.grantee`, `content_config_permissions.grantee_type`,
`notifications.completed`, `notifications.error`, `notifications.progressing`, `notifications.warning`,
`thumbnail_config.storage_class`, `thumbnail_config_permissions.access`, `thumbnail_config_permissions.grantee`, `thumbnail_config_permissions.grantee_type`.
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSElasticTranscoderPipeline_'
--- PASS: TestAccAWSElasticTranscoderPipeline_basic (151.20s)
--- PASS: TestAccAWSElasticTranscoderPipeline_kmsKey (263.72s)
--- PASS: TestAccAWSElasticTranscoderPipeline_notifications (384.76s)
--- PASS: TestAccAWSElasticTranscoderPipeline_withContentConfig (278.10s)
--- PASS: TestAccAWSElasticTranscoderPipeline_withPermissions (137.67s)
--- PASS: TestAccAWSElasticTranscoderPipeline_disappears (175.47s)
```

### CODE REVIEW

**Review by anGie44 — COMMENTED** [review]
Just a couple nitpicks here and comment re: the "grantee" validation. i'm not sure if we can readily validate all the possibilities, but we should definitely document them if not already done so 👍

File: aws/resource_aws_elastic_transcoder_pipeline.go [hunk 1]
```diff
@@ -64,15 +69,33 @@ func resourceAwsElasticTranscoderPipeline() *schema.Resource {
...
+									"ReadAcp",
+									"WriteAcp",
+									"FullControl",
+								}, false),
+							},
 						},
 						"grantee": {
 							Type:     schema.TypeString,
 							Optional: true,
+							ValidateFunc: validation.StringInSlice([]string{
```
[the remark this answers is not in the document]
Comment (DrFaust92) [reply aws/resource_aws_elastic_transcoder_pipeline.go:85]: Youre right, i missed that

**Comment by DrFaust92** [comment]
Added docs and addressed comments

**Review by anGie44 — APPROVED** [review]
Awesome, LGTM 👍 thanks @DrFaust92 

Output of acceptance tests (commercial):
```
--- PASS: TestAccAWSElasticTranscoderPipeline_disappears (19.42s)
--- PASS: TestAccAWSElasticTranscoderPipeline_kmsKey (21.90s)
--- PASS: TestAccAWSElasticTranscoderPipeline_withPermissions (23.32s)
--- PASS: TestAccAWSElasticTranscoderPipeline_notifications (35.24s)
--- PASS: TestAccAWSElasticTranscoderPipeline_withContentConfig (35.26s)
--- PASS: TestAccAWSElasticTranscoderPipeline_basic (39.11s)
```

**Pull request merged**